health/probe/plugin_impl_prometheus.go

```diff
@@ -124,6 +121,61 @@ func (p *PrometheusPlugin) AfterInit() error {
 		p.Log.Info("Unable to register Prometheus metrics handlers, HTTP is nil")
 	}
 
+	//TODO: Need improvement - instead of the exposing the map directly need to use in-memory mapping
+	if p.PluginStatusCheck != nil {
+		allPluginStatusMap := p.PluginStatusCheck.GetAllPluginStatus()
+		for k, v := range allPluginStatusMap {
+			p.Log.Infof("k=%v, v=%v, state=%v", k, v, v.State)
+			p.registerGauge(
+				Namespace,
+				Subsystem,
+				DependencyHealthName,
+				DependencyHealthHelp,
+				prometheus.Labels{
+					ServiceLabel:    p.getServiceLabel(),
+					DependencyLabel: k,
+				},
+				p.getDependencyHealth(k, v),
```

Comment:
Plugin status 'v' should not be collected here.  Rather, should be collected when getDependencyHealth() is invoked -- i.e., inside its body.